arrow/locales.py

```diff
@@ -1076,37 +1076,101 @@ class AzerbaijaniLocale(Locale):
     day_names = ['', 'Bazar ertəsi', 'Çərşənbə axşamı', 'Çərşənbə', 'Cümə axşamı', 'Cümə', 'Şənbə', 'Bazar']
     day_abbreviations = ['', 'Ber', 'Çax', 'Çər', 'Cax', 'Cüm', 'Şnb', 'Bzr']
 
-
-class ArabicLocale(Locale):
-
-    names = ['ar', 'ar_eg']
+class DefaultArabicLocale(Locale):
+    names = ['ar', 'ar_ae', 'ar_bh', 'ar_dj', 'ar_eg', 'ar_eh', 'ar_er',
+    'ar_km', 'ar_kw', 'ar_ly', 'ar_om', 'ar_qa', 'ar_sa', 'ar_sd', 'ar_so',
+    'ar_ss', 'ar_td', 'ar_ye']
 
     past = 'منذ {0}'
     future = 'خلال {0}'
 
     timeframes = {
         'now': 'الآن',
-        'seconds': 'ثوان',
+        'seconds': {
+            'double' : 'ثانيتين',
+            'ten' : '{0} ثوان',
+            'higher' : '{0} ثانية'
+        },
         'minute': 'دقيقة',
-        'minutes': '{0} دقائق',
+        'minutes': {
+            'double' : 'دقيقتين',
+            'ten' : '{0} دقائق',
+            'higher' : '{0} دقيقة'
+        },
         'hour': 'ساعة',
-        'hours': '{0} ساعات',
+        'hours': {
+            'double' : 'ساعتين',
+            'ten' : '{0} ساعات',
+            'higher' : '{0} ساعة'
+        },
         'day': 'يوم',
-        'days': '{0} أيام',
+        'days': {
+            'double' : 'يومين',
+            'ten' : '{0} أيام',
+            'higher' : '{0} يوم'
+        },
         'month': 'شهر',
-        'months': '{0} شهور',
+        'months': {
+            'double' : 'شهرين',
+            'ten' : '{0} أشهر',
+            'higher' : '{0} شهر'
+        },
         'year': 'سنة',
-        'years': '{0} سنوات',
+        'years': {
+            'double' : 'سنتين',
+            'ten' : '{0} سنوات',
+            'higher' : '{0} سنة'
+        },
     }
 
-    month_names = ['', 'يناير', 'فبراير', 'مارس', 'أبريل', 'مايو', 'يونيو', 'يوليو',
-        'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر']
-    month_abbreviations = ['', 'يناير', 'فبراير', 'مارس', 'أبريل', 'مايو', 'يونيو', 'يوليو',
-        'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر']
+    month_names = ['', 'يناير', 'فبراير', 'مارس', 'أبريل', 'مايو',
+        'يونيو', 'يوليو', 'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر']
+    month_abbreviations = ['', 'يناير', 'فبراير', 'مارس', 'أبريل', 'مايو',
+        'يونيو', 'يوليو', 'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر']
+
+    day_names = ['', 'الإثنين', 'الثلاثاء', 'الأربعاء', 'الخميس', 'الجمعة', 'السبت', 'الأحد']
+    day_abbreviations = ['', 'إثنين', 'ثلاثاء', 'أربعاء', 'خميس', 'جمعة', 'سبت', 'أحد']
+    
+    def _format_timeframe(self, timeframe, delta):
+        form = self.timeframes[timeframe]
+        delta = abs(delta)
+        if isinstance(form, dict):
+            if delta == 2:
+                form = form['double']
+            elif delta > 2 and delta <= 10:
+                form = form['ten']
+            else:
+                form = form['higher']
```

Comment:
Under this, a `delta` of 1 will get `form['higher']`.  Is that correct grammar?

Reply:
The assumption is that if its >1 `timeframe` will be `days`, `minutes`, etc. If delta is 1 then `timeframe` would be `day`, `minute`, etc. and correct grammar will be chosen. That assumption seems also to be true for other Locale implementations, including English.

Reply:
Makes sense. (I'm not as familiar with this area of the code.)